README.md

```diff
@@ -1,2 +1,209 @@
+
+A high-performance FastAPI service for generating text embeddings using SentenceTransformers, specifically designed for processing legal documents and search queries. The service efficiently handles both short search queries and lengthy court opinions, generating semantic embeddings that can be used for document similarity matching and semantic search applications. It includes support for GPU acceleration when available.
+
+The service is optimized to handle two main use cases:
+- Embedding search queries: Quick, CPU-based processing for short search queries
+- Embedding court opinions: GPU-accelerated processing for longer legal documents, with intelligent text chunking to maintain context
+
+## Features
+
+- Specialized text embedding generation for legal documents using the `sentence-transformers/multi-qa-mpnet-base-dot-v1`
+- Intelligent text chunking optimized for court opinions, based on sentence boundaries
+- Dedicated CPU-based processing for search queries, ensuring fast response times
+- GPU acceleration support for processing lengthy court opinions
+- Batch processing capabilities for multiple documents
+- Comprehensive text preprocessing and cleaning tailored for legal text
+- Health check endpoint
+
+## Installation
+
+This project uses UV for dependency management. To get started:
+
+1. Install [UV](https://docs.astral.sh/uv/getting-started/installation/):
+   ```bash
+   curl -LsSf https://astral.sh/uv/install.sh | sh
+   ```
+
+2. Clone the repository and install dependencies:
+   ```bash
+   git clone https://github.com/freelawproject/inception
+   cd inception
+   uv sync --extra cpu
+   ```
+   Use `--extra gpu`  for CUDA GPU support.
+
+## Quick Start
+
+### Running the Service
+
+The easiest way to run the embedding service is using Docker.
+
+Build:
+```bash
+docker build -t inception:latest --build-arg TARGET_ENV=prod .
+```
+
+Run:
+```bash
+docker run -d -p 8005:8005 inception
+```
+
+Run from hosted image:
+```bash
+docker run -d -p 8005:8005 freelawproject/inception:v2
+```
+
+For development run it with docker compose:
+```bash
+docker compose -f docker-compose.dev.yml up
+```
+
+To handle more concurrent tasks, increase the number of workers:
+```bash
+docker run -d -p 8005:8005 -e EMBEDDING_WORKERS=4 freelawproject/inception:v2
+```
+
+Test that the service is running:
+```bash
+curl http://localhost:8005
+# Should return: "Heartbeat detected."
+```
+
+### Using the Service
+
+You can interact with the service using any HTTP client. Here's a Python example using the `requests` library:
+
+```python
+import requests
+
+# Initialize base URL
+base_url = "http://localhost:8005"
+
+# Get embedding for a query
+response = requests.post(
+    f"{base_url}/api/v1/embed/query",
+    json={"text": "What is copyright infringement?"}
+)
+query_embedding = response.json()["embedding"]
+
+# Get embeddings for a document
+response = requests.post(
+    f"{base_url}/api/v1/embed/text",
+    data="The court finds that..."
+)
```

Comment:
I regret to note that you forgot the timeout parameter. :) 

We should be good examples, right?

Reply:
That's right, thanks for catching that. I've added a timeout to these examples and included an extra example for the batch processing endpoint.